waveorder/stokes.py

```diff
@@ -0,0 +1,338 @@
+                    -np.cos(chi),
+                ],
+            ]
+        )
+    else:
+        raise KeyError(f"{scheme} is not implemented, use 4-State or 5-State")
+    return A
+
+
+def s0123_CPL_after_ADR(ret, ori, tra, dop):
+    """
+    Returns the Stokes parameters of circularly polarized light (CPL) that
+    has passed through an attenuating depolarizing retarder (ADR) parametrized
+    by its retardance (ret), slow-axis orientation (ori), transmittance (tra),
+    and depolarization (dop).
+
+    Note: all four parameters can be ndarrays, but they must be the same size.
+    If your parameters are in an ndarray with array.shape = (4, ...), use
+    the * operator to expand over the first dimension.
+
+    e.g. s0123_CPL_after_ADR(*array) is identical to
+    s0123_CPL_after_ADR(array[0], array[1], array[2], array[3]).
+
+    Parameters
+    ----------
+    ret, ori, tra, dop : ArrayLike, float
+        ret: retardance of ADR, 2*pi periodic
+        ori: slow-axis orientation of ADR, 2*pi periodic
+        tra: transmittance of ADR, 0 <= tra <= 1
+        dop: depolarization of ADR, 0 <= dop <= 1
+
+    Returns
+    -------
+    s0, s1, s2, s3: ArrayLike, float
+        Stokes parameters
+
+    """
+    s0 = tra
+    s1 = tra * dop * np.sin(ret) * np.sin(2 * ori)
+    s2 = tra * dop * -np.sin(ret) * np.cos(2 * ori)
+    s3 = tra * dop * np.cos(ret)
+    return s0, s1, s2, s3
+
+
+def s012_CPL_after_AR(ret, ori, tra):
+    """
+    Returns the first three Stokes parameters of circularly polarized light
+    (CPL) that has passed through an attenuating retarder [AR] parametrized by
+    its retardance (ret), slow-axis orientation (ori), and transmittance (tra).
+
+    This model is used to model non-depolarizing samples, or situations where
+    depolarization information is unavailable...e.g. when only a linear Stokes
+    polarimeter is available.
+
+    Parameters
+    ----------
+    ret, ori, tra: ArrayLike, float
+        ret: retardance of ADR, 2*pi periodic
+        ori: slow-axis orientation of ADR, 2*pi periodic
+        tra: transmittance of ADR, 0 <= tra <= 1
+
+    Returns
+    -------
+    s0, s1, s2: ArrayLike, float
+        First three Stokes parameters
+
+    """
+    s0 = tra
+    s1 = tra * np.sin(ret) * np.sin(2 * ori)
+    s2 = tra * -np.sin(ret) * np.cos(2 * ori)
+    return s0, s1, s2
+
+
+# Inverse function group
+
+
+def _s12_to_ori(s1, s2):
+    """
+    Converts s1 and s1 into a slow-axis orientation.
+
+    This functions matches the sign convention used in s0123_CPL_after_ADR and
+    s012_CPL_after_AR (see tests for examples), and matches the orientation
+    range convention used in comp-micro: 0 <= ori < pi.
+
+    Parameters
+    ----------
+    s1, s2: ArrayLike, float
+        Stokes parameters
+
+    Returns
+    -------
+    ArrayLike, float
+        Slow-axis orientation with 0 <= ori < pi.
+    """
+    return (np.arctan2(s1, -s2) % (2 * np.pi)) / 2
```

Comment:
@Soorya19Pradeep asked about consistency of this convention with the existing `waveorder` implementation. 

**My convention here is not consistent with the existing `waveorder_reconstructor` implementation**, but it is consistent with the  e-life paper, with the wikipedia article on retarders, and (most importantly) self consistent within this module. 

I can change this convention to be backwards compatible, but we will lose consistency with the e-life paper and (less importantly) it will require changes across this module to maintain self consistency. 

We have already placed some of the burden of orientation conventions on the user now that we have the "90 degree shift" checkbox in recOrder, so I'm leaning towards sticking with the convention implemented here. But I'm interested in hearing your thoughts @mattersoflight.